src/calibration.c

```diff
@@ -492,22 +492,38 @@ void vmLoadOffringaSolution( vcsbeam_context *vm )
         fprintf( stderr, "than specified (%u)\n", nant );
         exit(EXIT_FAILURE);
     }
-    if (channelCount != nChan)
+
+    // Now set the calibration solution settings correctly
+    fprintf( stdout, "Calibration solution (%s) ", vm->cal.caldir );
+    fprintf( stdout, "has (%u) total channels\n", channelCount );
+    nChan = channelCount;
+
+    // Figure out the correct number of fine channels in a coarse channel from 
+    // the calibration solution
+    if (vm->cal.picket_fence)
     {
-        fprintf( stdout, "Warning: Calibration solution (%s) ", vm->cal.caldir );
-        fprintf( stdout, "contains a different number (%u) ", channelCount );
-        fprintf( stdout, "than the requested (%u) channels.\n", nChan );
-        nChan = channelCount;
-        nchan = nChan / vm->mpi_size;
-        interp_factor = vcs_nchan / nchan;
-        fprintf( stdout, "Assuming calibration channels are "
-                "%d kHz\n", vm->cal_metadata->coarse_chan_width_hz / nchan / 1000 );
+        // Assuming calibration solution has same number of coarse channels as 
+        // the number of MPI processes for picket fenced data
+        fprintf( stdout, "Assuming %u coarse channels in the calibration "
+                "solution for picket fence data\n", vm->mpi_size);
+        nchan = nChan / vm->mpi_size; 
+    }
+    else
+    {
+        // Assuming 24 coarse channels in calibration solution for contiguous 
+        // data
+        fprintf( stdout, "Assuming %u coarse channels in the calibration "
+                "solution for contiguous data\n", 24);
```

Comment:
There is a codacy warning on this line. I think just using %d instead of %u will fix it